decision_service/dsclientlib/error_callback_fn.cc

```diff
@@ -0,0 +1,36 @@
+#include "error_callback_fn.h"
+
+using namespace std;
+namespace decision_service {
+
+  void error_callback_fn::set(error_fn fn, void* context)
+  {
+    lock_guard<mutex> lock(_mutex);
```

Comment:
>lock_guard<mutex> lock(_mutex); [](start = 4, length = 31)

this is to make sure we have a consistent combination of fn/context?

Reply:
yes

---
In reply to: [186736980](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r186736980) [](ancestors = 186736980)